decisions/2024-04-15_decimal_pack_size.md

```diff
@@ -0,0 +1,131 @@
+- Straightforward for all reporting. What's in the invoice/requisition/PO/tender/GR is consistent with what the user entered.
+- Straightforward for integrations, no deep explanation needed.
+
+_Cons:_
+
+- UI needs to handle potentially large amounts of decimal digits gracefully. e.g. `1.3333333...` to `1.33` (2dp rounding, preference) or `1 1/3` (can be presented with neat HTML, but only nice for fractions with small denominators, harder to compare one cell to the next).
+- Backend should keep full stored value to maximum accuracy. These values can be smoothed over in some cases (e.g 0.9999999999999997 can be rounded up and treated as 1 in UI, in the backend it'll be that quantity).
+- Calculations will accrue inaccuracy. For individual lines this is usually tolerable but it does accumulate! In tenders we see it in monetary values: summing the "adjust cost" of several thousand lines can result in whole dollars of inaccuracy. Governments/Procurement agencies may have policies of acceptable level of calculation inaccuracy.
+  - It is worth highlighting that number of packs and our monetary fields are all already floating point type with these same problems, so we must already handle all the above issues regardless. **So it's not really a con if we have to do it anyway.** It just makes the surface area of floating point accuracy a smidgen higher.
+- recurring numbers create headaches of who truly has the "full" amount of something. E.g
+
+### Option 2 - Round pack size up to next whole number and adjust number of packs, monetary fields etc.
+
+We convert decimal pack sizes up to the next whole number.
+
+E.g.
+
+1 pack of 1L is repacked to 40 packs of 0.05L in mSupply desktop. The latter on integration in OMS:
+
+1. pack size: 0.05L/0.05 = 1L
+2. number of packs: 40packs\*0.05 = 1pack
+3. cost/sell price: 0.5/0.05 = $10
+
+The pharmacist feedback I have received on this is that it'd be confusing to users if they had a system telling them they had 1 of when they can see there are 40 smaller vessels on the shelf. For accounting, I have not asked a finance stakeholder yet but I presume this is very very bad form in regards of adjusting the number of what was physically ordered/received and the cost of each item ordered.
+
+A potential solution for this is to use pack variants. After converting back to 1L pack size, we use a variant of "50mL bottle" that converts all the properties back in the UI layer. Pack variants don't appear to quite support this - the assumption is the base unit is a small whole number (integer). In this case we'd need a pack variant of 0.05L, which would do the reverse of the calculations above. There is additionally an issue of when and how these pack variants are generated, as they are centrally controlled data.
+
+Going to the option 1 problem statement, we also have to solve this at integration still so no avoided in anyway:
```

Comment:
I added a comment about the pack size never being repeating decimal, does that help with resolving this ?